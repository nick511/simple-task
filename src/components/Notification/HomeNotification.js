import React from 'react'
import PropTypes from 'prop-types'

import Notification, { TYPE_INFO, TYPE_SUCCESS } from './Notification'

const HomeNotification = ({ tasks }) => {
  const completedTasks = tasks.filter(t => !t.isCompleted).length
  let notiTittle = 'Complete all tasks'
  let notiDesc = `You have ${completedTasks} active tasks`
  let notiType = TYPE_INFO
  let notiProgress = { max: tasks.length, value: tasks.length - completedTasks }

  if (tasks.every(t => t.isCompleted)) {
    notiTittle = 'All tasks completed'
    notiDesc = 'Well done!'
    notiType = TYPE_SUCCESS
    notiProgress = null
  }

  return (
    <Notification title={notiTittle} desc={notiDesc} type={notiType} progress={notiProgress} />
  )
}

HomeNotification.propTypes = {
  tasks: PropTypes.array.isRequired,
}

export default HomeNotification
