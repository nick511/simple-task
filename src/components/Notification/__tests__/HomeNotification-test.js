import React from 'react'
import { shallow } from 'enzyme'

import HomeNotification from '../HomeNotification'

function getTestTasks (num = 2) {
  return Array(num).fill().map((e, i) => (
    {
      id: i,
      name: 'test' + i,
      actions: {
        updateStatus: jest.fn(),
        updateName: jest.fn(),
      }
    }
  ))
}

function getWrapper (tasks = null) {
  return shallow(<HomeNotification tasks={tasks || getTestTasks()} />)
}

describe('<HomeNotification />', () => {

  test('has 2 active tasks', () => {
    const wrapper = getWrapper()
    expect(wrapper.find('Notification').props()).toMatchSnapshot()
  })

  test('has all tasks completed', () => {
    const tasks = getTestTasks(2).map(task => {
      task.isCompleted = true
      return task
    })
    const wrapper = getWrapper(tasks)
    expect(wrapper.find('Notification').props()).toMatchSnapshot()
  })
})
