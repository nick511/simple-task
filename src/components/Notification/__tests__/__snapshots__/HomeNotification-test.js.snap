// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeNotification /> has 2 active tasks 1`] = `
Object {
  "desc": "You have 2 active tasks",
  "progress": Object {
    "max": 2,
    "value": 0,
  },
  "title": "Complete all tasks",
  "type": "is-info",
}
`;

exports[`<HomeNotification /> has all tasks completed 1`] = `
Object {
  "desc": "Well done!",
  "progress": null,
  "title": "All tasks completed",
  "type": "is-success",
}
`;
